rules/community/gcp/gcp_kms_decryption_by_unexpected_service_account.yaral

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+rule gcp_kms_decryption_by_unexpected_service_account {
+  meta:
+    author = "Drew Pilarski - Tempus AI"
+    description = "Detects when a Key Management Service (KMS) decryption operation is carried out by a service account that's not in the allowlist. This behavior may indicate unauthorized access to information."
+    rule_id = "mr_90cd5b0d-60a7-4028-bc41-dce0947fbafd"
+    rule_name = "Unauthorized KMS Decryption"
+    reference = "Internal Security Policy" // Include link to your organization's internal documentation if available
+    mitre_attack_tactic = "Persistence"
+    mitre_attack_technique = "Valid Accounts: Cloud Accounts"
+    type = "Alert"
+    platform = "GCP"
+    data_source = "Cloud Audit Logs"
+    severity = "Medium"  // Adjust based on your risk assessment
+    priority = "Medium"  // Adjust based on your incident response process
+
+  events:
+
+    ( // Generic KMS Key Pattern - Pick the relevant one to your environment

```

Comment:
```suggestion
  events:
    $decrypt.metadata.product_name = "Google Cloud Platform"
    $decrypt.metadata.vendor_name = "Google Cloud Platform"

    ( // Generic KMS Key Pattern - Pick the relevant one to your environment
```

Reply:
I'd recommend adding `metadata.product_name` and `metadata.vendor_name`. It'll filter down the number of events before the rest of your rule logic is applied. The rule will be more efficient and performant.

`metadata.event_type` and `metadata.product_event_type` can also be good ones to use early on in the events section.

Reply:
I attempted to ingest the logs that you attached to #111, but I'm having some trouble with that. I'm wondering if a sanitized field/value is preventing me from ingesting them. After you've made the above changes to the rule, can you please validate that the rule matches on the events in your tenant? Cheers.